plugins/modules/database.ps1

```diff
@@ -0,0 +1,244 @@
+        state = @{type = 'str'; required = $false; default = 'present'; choices = @('present', 'absent') }
+    }
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$sqlInstance = $module.Params.sql_instance
+$sqlUsername = $module.Params.sql_username
+if ($null -ne $sqlUsername) {
+    [securestring]$secPassword = ConvertTo-SecureString $module.Params.sql_password -AsPlainText -Force
+    [pscredential]$sqlCredential = New-Object System.Management.Automation.PSCredential ($sqlUsername, $secPassword)
+}
+$database = $module.Params.database
+$recoveryModel = $module.Params.recovery_model
+$dataFilePath = $module.Params.data_file_path
+$logFilePath = $module.Params.log_file_path
+$ownerName = $module.Params.owner_name
+$compatibility = $module.Params.compatibility
+[nullable[bool]]$rcsiEnabled = $module.Params.rcsi
+[nullable[int]]$maxDop = $module.Params.maxdop
+[nullable[int]]$secondaryMaxDop = $module.Params.secondary_maxdop
+$state = $module.Params.state
+$checkMode = $module.CheckMode
+
+# Get database status
+try {
+    $server = Connect-DbaInstance -SqlInstance $sqlInstance -SqlCredential $sqlCredential
+    $getDatabaseHash = @{
+        SqlInstance = $sqlInstance
+        SqlCredential = $sqlCredential
+        Database = $database
+        OnlyAccessible = $true
+        ExcludeSystem = $true
+        EnableException = $true
+    }
+    $output = Get-DbaDatabase @getDatabaseHash
+}
+catch {
+    $module.FailJson("Error checking database status.", $_.Exception.Message)
+}
+
+if ($state -eq "absent") {
+    if ($null -ne $output) {
+        $dropHash = @{
+            SqlInstance = $sqlInstance
+            SqlCredential = $sqlCredential
+            Database = $database
+            EnableException = $true
+            Confirm = $false
+        }
+        Remove-DbaDatabase @dropHash
+        $module.Result.changed = $true
+    }
+    #TODO: Is it wrong to not return a useless dictionary of dropped db
+    # attributes here? It makes different states = different output values.
+    # or maybe default to a few basic properties (see memory module's output)
+    $module.ExitJson()
+}
+elseif ($state -eq "present") {
+    # Create database
+    if ($null -eq $output) {
+        try {
+            $newDbParams = @{
+                SqlInstance = $sqlInstance
+                SqlCredential = $sqlCredential
+                Database = $database
+                Owner = $OwnerName
+                EnableException = $true
+            }
+            if ($null -ne $dataFilePath) {
+                $newDbParams.Add("DataFilePath", $dataFilePath)
+            }
+            if ($null -ne $logFilePath) {
+                $newDbParams.Add("LogFilePath", $logFilePath)
+            }
+            $output = New-DbaDatabase @newDbParams
+            $module.Result.changed = $true
+        }
+        catch {
+            $module.FailJson("Creating database [$database] failed.", $_.Exception.Message)
```

Comment:
```suggestion
            $module.FailJson("Creating database [$database] failed.", $_)
```
I believe this second parameter should be the whole exception/errorrecord object.